packages/x-components/cypress.config.ts

```diff
@@ -0,0 +1,42 @@
+import { addCucumberPreprocessorPlugin } from '@badeball/cypress-cucumber-preprocessor';
+import createEsbuildPlugin from '@badeball/cypress-cucumber-preprocessor/esbuild';
+import { createBundler } from '@bahmutov/cypress-esbuild-preprocessor';
+import { defineConfig } from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    baseUrl: 'http://localhost:8080',
+    defaultCommandTimeout: 7000,
+    viewportHeight: 1080,
+    viewportWidth: 1920,
+    screenshotOnRunFailure: false,
+    video: false,
+    supportFile: 'tests/support/index.ts',
+    fixturesFolder: 'tests/e2e/fixtures',
+    screenshotsFolder: 'tests/e2e/screenshots',
+    retries: {
+      openMode: 0,
+      runMode: 1
+    },
+    async setupNodeEvents(on, config) {
+      await addCucumberPreprocessorPlugin(on, config);
+      on(
+        'file:preprocessor',
+        createBundler({
+          plugins: [createEsbuildPlugin(config)]
```

Comment:
The recommended option  with the new cypress-cucumber-preiprocessor is to use esbuild (I won't complain).